docker/pypi/wmagent/run.sh

```diff
@@ -0,0 +1,683 @@
+    local certMinLifetimeSec=$(($certMinLifetimeHours*60*60))
+    # DONE: Here to find out if the agent is CERN or FNAL and change renew_proxy.sh respectively
+    if [[ "$HOSTNAME" == *cern.ch ]]; then
+        local myproxyCredName="amaltaroCERN"
+    elif [[ "$HOSTNAME" == *fnal.gov ]]; then
+        local myproxyCredName="amaltaroFNAL"
+    else
+        echo "$FUNCNAME: ERROR: Sorry, we do not recognize the network domain name of the current host: $HOSTNAME"
+        return $(false)
+    fi
+    sudo sed -i "s/credname=CREDNAME/credname=$myproxyCredName/g" $WMA_DEPLOY_DIR/deploy/renew_proxy.sh
+    sudo chmod 755 $WMA_DEPLOY_DIR/deploy/renew_proxy.sh
+
+    # Here to check certificates and update myproxy if needed:
+    if [[ -f $WMA_CERTS_DIR/servicecert.pem ]] && [[ -f $WMA_CERTS_DIR/servicekey.pem ]]; then
+
+        echo "$FUNCNAME: Checking Certificate lifetime:"
+        local now=$(date +%s)
+        local certEndDate=$(openssl x509 -in $WMA_CERTS_DIR/servicecert.pem -noout -enddate)
+        certEndDate=${certEndDate##*=}
+        echo "$FUNCNAME: Certifficate end date: $certEndDate"
+        [[ -z $certEndDate ]] && { echo "ERROR: Failed to determine certificate end date!"; return $(false) ;}
+
+        certEndDate=$(date --date="$certEndDate" +%s)
+        [[ $certEndDate -le $now ]] && { echo "ERROR: Expired certificate at $WMA_CERTS_DIR/servicecert.pem!"; return $(false) ;}
+        [[ $(($certEndDate -$now)) -le $certMinLifetimeSec ]] && { echo "WARNING: The service certificate lifetime is less than certMinLifetimeHours: $certMinLifetimeHours! Please update it ASAP!" ;}
+
+        # Renew myproxy if needed:
+        echo "$FUNCNAME: Checking myproxy lifetime:"
+        local myproxyEndDate=$(openssl x509 -in $WMA_CERTS_DIR/myproxy.pem -noout -enddate)
+        myproxyEndDate=${myproxyEndDate##*=}
+        echo "$FUNCNAME: myproxy end date: $myproxyEndDate"
+        [[ -n $myproxyEndDate ]] || ($WMA_DEPLOY_DIR/deploy/renew_proxy.sh) || { echo "ERROR: Failed to renew invalid myproxy"; return $(false) ;}
+
+        myproxyEndDate=$(date --date="$myproxyEndDate" +%s)
+        [[ $myproxyEndDate -gt $now ]] || ($WMA_DEPLOY_DIR/deploy/renew_proxy.sh) || { echo "ERROR: Failed to renew expired myproxy"; return $(false) ;}
+
+        # Stay safe and always change the service {cert,key} and myproxy mode here:
+        sudo chmod 600 $WMA_CERTS_DIR/*
+        echo "$FUNCNAME: OK"
+    else
+        echo "ERROR: We found no service certificate installed at $WMA_CERTS_DIR!"
+        echo "ERROR: Please install proper cert and key files before restarting the WMAgent container!"
+        return $(false)
+    fi
+
+    # Update flavor/type/domain global variables if needed
+    # (grep -E "^[[:blank:]]*ORACLE_" $WMA_ADMIN_DIR/WMAgent.secrets > /dev/null) && CURR_FLAVOR=oracle || CURR_FLAVOR=mysql
+
+    echo "Done: $stepMsg"
+    echo "-------------------------------------------------------"
+}
+
+_check_oracle()
```

Comment:
This function is actually arguable. We will be running new containers whenever a new WMAgent release comes out, so Oracle database will not be empty in that scenario.
Maybe we simply check and inform what the status is?

Reply:
done